helpers/mathjax.js

```diff
@@ -60,32 +60,25 @@ export class HtmlBlockMathRenderer {
 		// This work-around should be removed when linebreaks are natively supported.
 		// MathJax issue: https://github.com/mathjax/MathJax/issues/2312
 		// A duplicate that explains our exact issue: https://github.com/mathjax/MathJax/issues/2495
-		const lineBreakStyle = 'display: block; height: 0.5rem;';
-
-		// If we're opting out of deferred rendering, we need to rely
-		// on the global MathJax install for rendering.
-		if (options.noDeferredRendering) {
-			elem.querySelectorAll('mspace[linebreak="newline"]').forEach(elm => {
-				elm.setAttribute('style', lineBreakStyle);
-			});
-
-			await window.MathJax.startup.promise;
-			window.MathJax.typesetShadow(elem.getRootNode(), elem);
-			return elem;
-		}
+		elem.querySelectorAll('mspace[linebreak="newline"]').forEach(elm => {
+			elm.setAttribute('style', 'display: block; height: 0.5rem;');
+		});
+
+		await window.MathJax.startup.promise;
+		await window.MathJax.typesetPromise([elem]);
 
-		const inner = elem.innerHTML.replace(/<mspace linebreak="newline">/gi, `<mspace linebreak="newline" style="${lineBreakStyle}">`);
+		// If we're opting out of deferred rendering, we can rely
+		// on the global MathJax styles for rendering.
+		if (options.noDeferredRendering) return elem;
```

Comment:
Hmm. I suspect I need to do something more clever here. If we're in a nested shadowDOM this will break because the styles won't get set properly.